var/spack/repos/builtin/packages/intel-tbb/package.py

```diff
@@ -125,7 +125,7 @@ class IntelTbb(CMakePackage, MakefilePackage):
     patch("gcc_generic-pedantic-4.4.patch", level=1, when="@:2019.0")
 
     # Patch and conflicts for GCC 13 support (#1031).
-    patch("gcc_13-2021.patch", when="@2021.1:")
+    patch("gcc_13-2021.patch", when="@2021.1: %gcc")
```

Comment:
```suggestion
    patch("gcc_13-2021.patch", when="@2021.1: 2021.9")
```

Would this work for macos? I believe that if gcc needs the patch, then %intel and %oneapi need it too because they use the same header files. 2021.10.0 upstream is fixed and does not need the patch. If the patch succeeds, then including cstdlib should be harmless.

Reply:
I think the issue is that Apple's `/usr/bin/patch` (`2.0-12u11-Apple`) fails to apply the patch. With `gpatch` installed via homebrew (`GNU patch 2.7.6`), applying the patch succeeds for me.

Instead of restricting the patch to %gcc, I would like to propose to fix the patch in this PR: #40640.